djangocms_fil_bootstrap/components/collections.py

```diff
@@ -9,51 +9,25 @@ class Collections(Component):
     default_factory = dict
 
     def parse(self):
-        if not self.raw_data.get("enable", False):
-            return
-        moderator = self.bootstrap.users["moderator"]
-        moderator2 = self.bootstrap.users["moderator2"]
-        moderator3 = self.bootstrap.users["moderator3"]
-        collection1 = ModerationCollection.objects.create(
-            author=moderator,
-            name="Collection 1",
-            workflow=self.bootstrap.workflows["wf4"],
-        )
-        collection1.add_version(get_version(self.bootstrap.pages["page1"]))
-        collection1.add_version(get_version(self.bootstrap.pages["page2"]))
-
-        collection2 = ModerationCollection.objects.create(
-            author=moderator2,
-            name="Collection 2",
-            workflow=self.bootstrap.workflows["wf4"],
-        )
-        collection2.add_version(get_version(self.bootstrap.pages["page3"]))
-        collection2.add_version(get_version(self.bootstrap.pages["page4"]))
-
-        collection3 = ModerationCollection.objects.create(
-            author=moderator3,
-            name="Collection 3",
-            workflow=self.bootstrap.workflows["wf4"],
-        )
-        collection3.add_version(get_version(self.bootstrap.pages["page5"]))
-
-        collection4 = ModerationCollection.objects.create(
-            author=moderator,
-            name="Collection 4",
-            workflow=self.bootstrap.workflows["wf4"],
-        )
-        collection4.add_version(get_version(self.bootstrap.pages["page6"]))
-
-        collection5 = ModerationCollection.objects.create(
-            author=moderator2,
-            name="Collection 5",
-            workflow=self.bootstrap.workflows["wf4"],
-        )
-        collection5.add_version(get_version(self.bootstrap.pages["page7"]))
-
-        collection6 = ModerationCollection.objects.create(
-            author=moderator3,
-            name="Collection 6",
-            workflow=self.bootstrap.workflows["wf5"],
-        )
-        collection6.add_version(get_version(self.bootstrap.pages["page8"]))
+        for name, data in self.raw_data.items():
+            collection_data = {
+                "author": self.bootstrap.users[data["user"]],
+                "workflow": self.bootstrap.workflows[data["workflow"]],
+                "name": data["name"],
+            }
+            collection = self.get_or_create(collection_data)
```

Comment:
`parse` needs to have a test which checks what happens if a ModerationCollection like this already exists. Also, unless implementation changes, a test for what if the author matches but other two params don't, what if a workflow matches but the other params don't and what if the name matches but the other two params don't.
Incidentally, at current, the way this is written I guess all three would need to match for a ModerationCollection to not be created. Is that correct? And if so, shouldn't this be documented? It doesn't seem obvious to me. I'd probably assume the name would be the only identifier tbh.

Reply:
This is already tested in `test_get_or_create`. I have added documentation as you suggest.